src/main/java/frc/robot/utils/diag/DiagClimber.java

```diff
@@ -0,0 +1,48 @@
+package frc.robot.utils.diag;
+
+import com.revrobotics.CANSparkMax;
+
+public class DiagClimber extends DiagSparkMaxEncoder {
+    private double requiredRaise;
+    private double requiredLower;
+    private boolean loweredTrue;
+    private boolean raisedTrue;
+    public DiagClimber(String title, String name, double requiredTravel, CANSparkMax canSparkMax, double requiredLower) {
+        super(title, name, requiredTravel, canSparkMax);
+        this.requiredLower = requiredLower;
+        this.requiredRaise = requiredTravel;
+        reset();
+    }
+    public boolean getDiagResultRaised() {
```

Comment:
Instead of calling these `getDiag...()` within the subsystem directly, methods should be kept private. Instead, override the `setShuffleBoardTab()`, `reset()`, and `refresh()` methods to add the additional shuffleboard values for displaying the raised and lowered values. Then call `Robot.getDiagnostics().addDiagnosable()` within the subsystem.